src/seraphis_main/scan_balance_recovery_utils.cpp

```diff
@@ -308,18 +308,19 @@ static std::unordered_set<rct::key> process_chunk_sp_selfsend_pass(
         for (const ContextualBasicRecordVariant &contextual_basic_record :
             chunk_basic_records_per_tx.at(tx_with_spent_enotes))
         {
-            if (!contextual_basic_record.is_type<SpContextualBasicEnoteRecordV1>())
+            const auto * const sp_contextual_basic_record{
+                    contextual_basic_record.try_unwrap<SpContextualBasicEnoteRecordV1>()
+                };
+            if (nullptr == sp_contextual_basic_record)
```

Comment:
I like using `<const value> == <variable>` because you will get a compiler error if you accidentally write `<const value`> = <variable>`, whereas if you accidentally write `<variable> = <const value>`, you just get weird runtime behavior that is hard to debug. TBF, `sp_contextual_basic_record` is a `const` variable here, so my mind could be changed.